cdk/src/open-data-platform/data-plane/schema/schema.sql

```diff
@@ -151,6 +156,95 @@ CREATE TABLE IF NOT EXISTS zipcodes (
 CREATE INDEX IF NOT EXISTS geom_index ON counties USING GIST (geom);
 CREATE UNIQUE INDEX IF NOT EXISTS zipcode_index ON zipcodes (zipcode);
 
+--- Returns lead connections aggregated by either state or water system, depending on z (zoom level).
+
+CREATE OR REPLACE FUNCTION public.lead_connections_function_source(z integer, x integer, y integer, query_params json) RETURNS bytea AS $$
+DECLARE
+    mvt bytea;
+BEGIN
+    IF (z < 6) THEN
+        -- Aggregate est. lead connections by state. Select relevant fields from both tables.
+        SELECT INTO mvt ST_AsMVT(tile,
+                                 'public.lead_connections_function_source',
+                                 4096, 'geom')
+        FROM (
+                 SELECT ST_AsMVTGeom(ST_Transform(s.geom, 3857),
+                                     ST_TileEnvelope(z, x, y)) AS geom,
+                        s.name                                 as state_name,
+                        count(w.lead_connections_count)        as lead_connections_count,
+                        count(w.service_connections_count)     as service_connections_count,
+                        count(w.population_served)             as population_served
+                 FROM water_systems w
+                          RIGHT JOIN states s
+                                     ON w.state_census_geo_id = s.census_geo_id
+                 WHERE ST_Transform(s.geom, 3857) && ST_TileEnvelope(z, x, y)
+                 GROUP BY s.geom, s.name
+             ) as tile
+        WHERE geom IS NOT NULL;
+    ELSE
+        -- Select water system fields.
+        SELECT INTO mvt ST_AsMVT(tile,
+                                 'public.lead_connections_function_source',
+                                 4096, 'geom')
+        FROM (
+                 SELECT ST_AsMVTGeom(ST_Transform(w.geom, 3857),
+                                     ST_TileEnvelope(z, x, y)) AS geom,
+                        w.pws_id                               as pws_id,
+                        w.pws_name                             as pws_name,
+                        w.lead_connections_count               as lead_connections_count,
+                        w.population_served                    as population_served
+                 FROM water_systems w
+                 WHERE ST_Transform(w.geom, 3857) && ST_TileEnvelope(z, x, y)
+                 GROUP BY w.geom, w.pws_id, w.pws_name,
+                          w.lead_connections_count, w.population_served
```

Comment:
Grouping by counts like this looks weird to me since these aren't really identifying properties. Replacing the columns with something like this might achieve the same result:
```
sum(w.lead_connections_count) as lead_connections_count,
sum(w.population_served) as population_served
```

Reply:
Done.